### ISSUE TITLE
[US_ID] (Day Zero) Overdue for discharge filters don't catch LIFE, may be getting distorted during ingest

### ISSUE THREAD

**Issue by cawarren** [Recidiviz/pulse-dashboard#1350]
@ohaibbq @Catacola 
(CC @DSharm only because you helped set build this view early on)
(CC @colincadams in case related to DQ)

We should verify that when the full-term release date (FTRD) is LIFE (and 09/09/9999, and 99/99/9999, and 09/09/1999), those are excluded from 'Overdue for discharge'.

For examples, [here's the spreadsheet](https://docs.google.com/spreadsheets/d/1hFEJS_EF9zlzZOtvqAA6UOYzKzHPcdbd/edit#gid=795539973) we used for validation of the list yesterday:
- E.g., row 124 (our data shows 1999, theirs 9999...is this a data quality or ingest issue?)
- On some cases w/really old release dates, we’re wrong, and we’ll see ‘Activation’ or ‘Court order’ in case updates, indicating new case was filed, charges brought, etc. E.g., row 122 (our projected end date was 10/26/1993, actual FTRD is 9/15/2023)
- Similar situation in row 133, our projected end-date is 6/23/2009; actual FTRD is 1/21/2023. In case updates, they have Case Activation and Warrant, suggesting this person got a new sentence.
- As an aside: PO’s don’t trust CIS’s FTRD; Reflections is technically the source of truth / might be more accurate, and frequently CIS has no FTRD and we do, because we got it from Reflections. But just in case, we may want to double-check that our logic is preferencing Reflections FTRD above CIS in cases where they both have it, but disagree.

**Comment by jkgerig** [comment]
regarding row 122 above, we appear to be missing a row in Idaho raw data table `sentence` for `docno = 33919`.

```
SELECT *
FROM `recidiviz-123.us_id_raw_data_up_to_date_views.sentence_latest`
INNER JOIN `recidiviz-123.us_id_raw_data_up_to_date_views.mittimus_latest`
    USING (mitt_srl)
WHERE docno = '33919'
ORDER BY 1, 2, 3
```

IDOC data systems show two records for the above query, while we only show 1. The `mitt_srl` of the missing row is `292778` and has a `sent_beg_dtd = 09/16/20` and `sent_ft_dtd = 09/15/23`.

**Comment by colincadams** [comment]
Currently our ingest only looks at the `is_lifer` flag from the [sentence table](https://app.gitbook.com/@recidiviz/s/recidiviz/state-ingest-catalog/us_id/raw_data/sentence). It also uses the dates from that table, not the mittimus table. It seems like we should at least fallback to looking at the date from the mittimus table to determine if something is a life sentence?

@jkgerig and I took a looked into the 9/9/99 case specifically yesterday. It looks like on the IDOC side they are sometimes storing the date as "9999" to indicate a life sentence, which then gets exported in the CSV on their side as "9/9/99". This is unfortunate as it parses for us as September 9th, 1999. The best solution would be if they exported a date with a 4 digit year so that we could differentiate this from the real date. In the meantime we can parse all "9/9/99" dates as life sentences. This does mean that for any historical cases where the actual full term release date was sept 9th 1999 we will mark this as a life sentence, but the error should be much lower.

**Comment by colincadams** [comment]
@jkgerig to ask Michele whether we should be using Mittimus or Sentence or some combination
@colincadams to pull some examples of discrepancies

**Comment by jkgerig** [comment]
tl;dr - we should use `sentence`.

A few notes/comments - it looks like however we are receiving the `sentence` table seems to be working, because I don't find any mismatches between our version of the `sentence` table and a static dump I just pulled from IDOC systems (see sheet [here](https://docs.google.com/spreadsheets/d/1rPDkMPwM3H0mupf57w-UlROmUloFrkQ_bZqw9PScno4/edit#gid=1940864071)).

This is good news, because it sounds from IDOC that the `sentence` table should be relied on more than `mittimus` for these types of questions. In the sheet I have linked, there are a few instances where `mittimus` does disagree and would indicate a life sentence where `sentence` doesn't. These three instances are highlighted on the sheet. Two of the people are currently at liberty, while the third is incarcerated with a `TERMER` status (general incarceration term).

My query for reference:
```
WITH sentence_data AS (
    SELECT *
    FROM (
        SELECT
            m.docno,
            m.mitt_srl,
            s.sent_no,
            CASE WHEN m.sent_exp_dtd = '1999-09-09 00:00:00' OR m.sent_exp_dtd = '9999-09-09 00:00:00' THEN 1 ELSE 0 END AS mitt_life_date,
            CASE WHEN s.sent_ft_dtd = '1999-09-09 00:00:00' OR s.sent_ft_dtd = '9999-09-09 00:00:00' THEN 1 ELSE 0 END AS sent_life_date,
            m.sent_eff_dtd AS mitt_beg_date,
            m.sent_exp_dtd AS mitt_end_date,
            s.sent_beg_dtd AS sent_beg_date,
            s.sent_ft_dtd AS sent_end_date,
            s.sent_min_yr,
            s.sent_min_mo,
            s.sent_min_da,
            s.sent_max_yr,
            s.sent_max_mo,
            s.sent_max_da,
        FROM `recidiviz-123.us_id_raw_data_up_to_date_views.sentence_latest` s
        INNER JOIN `recidiviz-123.us_id_raw_data_up_to_date_views.mittimus_latest` m
            USING (mitt_srl)
        WHERE lifer IS NULL -- exclude known life sentences
            AND s.sent_disp != 'A' -- exclude original sentences when they have been amended (the old/outdated record has sent_disp = A)
    )
    WHERE mitt_life_date = 1 OR sent_life_date = 1
),
idoc_data AS (
    SELECT
        CAST(docno AS string) AS docno,
        CAST(mitt_srl AS string) AS mitt_srl,
        CAST(sent_no AS string) AS sent_no,
        mitt_end_date AS idoc_mitt_end_date,
        sent_end_date AS idoc_sent_end_date,
        probable_life,
    FROM `recidiviz-staging.analyst_data_scratch_space.idoc_life_sentences_20211105`
    WHERE lifer IS NULL
)

SELECT
    s.*,
    d.idoc_mitt_end_date,
    d.idoc_sent_end_date,
    CASE WHEN CAST(CAST(s.mitt_end_date AS DATETIME) AS DATE) = d.idoc_mitt_end_date THEN 1 ELSE 0 END AS mitt_match,
    CASE WHEN CAST(CAST(s.sent_end_date AS DATETIME) AS DATE) = d.idoc_sent_end_date THEN 1 ELSE 0 END AS sent_match,    
    d.probable_life,
FROM sentence_data s
LEFT JOIN idoc_data d
    USING (docno, mitt_srl, sent_no)
ORDER BY 1, 2, 3
```

My query from IDOC systems:
```
WITH sentence_data AS (
	SELECT
		m.docno,
		s.mitt_srl,
		s.sent_no,
		s.lifer,
		CASE WHEN CAST(m.sent_exp_dtd AS date) = '9999-09-09' THEN 1 ELSE 0 END AS mitt_life,
		CASE WHEN CAST(m.sent_exp_dtd AS date) = '1999-09-09' THEN 1 ELSE 0 END AS mitt_life_false_positive,
		CASE WHEN CAST(s.sent_ft_dtd AS date) = '9999-09-09' THEN 1 ELSE 0 END AS sent_life,
		CASE WHEN CAST(s.sent_ft_dtd AS date) = '1999-09-09' THEN 1 ELSE 0 END AS sent_life_false_positive,
		CAST(m.sent_efF_dtd AS date) AS mitt_beg_date,
		CAST(m.sent_exp_dtd AS date) AS mitt_end_date,
		CAST(s.sent_beg_dtd AS date) AS sent_beg_date,
		CAST(s.sent_ft_dtd AS date) AS sent_end_date,
		s.sent_min_yr,
		s.sent_min_mo,
		s.sent_min_da,
		s.sent_max_yr,
		s.sent_max_mo,
		s.sent_max_da
	FROM DataWarehouse.dbo.mittimus m
	INNER JOIN DataWarehouse.dbo.sentence s
		ON m.mitt_srl = s.mitt_srl
	WHERE s.sent_disp <> 'A' -- exclude the original version of amended sentences
)

SELECT *, CASE WHEN lifer IS NOT NULL OR mitt_life = 1 OR sent_life = 1 THEN 1 ELSE 0 END AS probable_life
FROM sentence_data
WHERE lifer IS NOT NULL
	OR mitt_life = 1
	OR sent_life = 1
	OR mitt_life_false_positive = 1
	OR sent_life_false_positive = 1
ORDER BY docno, mitt_srl, sent_no
```

**Comment by colincadams** [comment]
Any outstanding work will be tracked in Recidiviz/recidiviz-data#9971
